helpers/elixir/bin/check_update.exs

```diff
@@ -1,31 +1,70 @@
-[dependency_name] = System.argv()
+defmodule UpdateChecker do
+  def run(dependency_name, credentials) do
+    set_credentials(credentials)
 
-# dependency atom
-dependency = String.to_atom(dependency_name)
+    # Update the lockfile in a session that we can time out
+    task = Task.async(fn -> do_resolution(dependency_name) end)
+    Task.await(task, 30000)
```

Comment:
@stevedomin - I've moved this to a task that gets timed out because I couldn't figure out a way to stop [this prompt](https://github.com/hexpm/hex/blob/253418deab14e42862fe080a76a3804d671e4d5b/lib/hex/remote_converger.ex#L337-L339) from causing an indefinite timeout in the case where no creds have been provided. Is there a better solution?

Reply:
No, that seems like the best way until Hex adds a non-interactive mode.

Reply:
Awesome. Any idea if they've got plans to?

Reply:
Not that I know of.

@ericmj - Hello! Would you be willing to accept a PR that adds a non-interactive mode to Hex? This would be super useful for automated tools like https://github.com/dependabot

Reply:
A non-interactive mode will probably not be possible because we don't want to be tied to an API that we need to be backwards compatible with. Also as a heads up everything under the `Hex` namespace is private API and `Mix.Dep.Fetcher` is also private API so expect it to break at any time.

We are working on https://github.com/hexpm/hex_erl that will eventually include things like dependency resolution but I am not sure if it will include all your use cases.

Reply:
Thanks @ericmj. The timeout approach here will work for us for now.

Out of interest, what do you see CI servers doing when a Mixfile has a private organisation but no credentials have been specified? Are they also implementing a timeout or catching the prompt somehow?

On the private API usage - Dependabot is trying to do dependency resolution and lockfile generation with as little additional work as possible here (i.e., without doing an install). Are there any plans to add a public API for that, similar to `cargo update` or `npm install --package-lock-only`?

Reply:
> Out of interest, what do you see CI servers doing when a Mixfile has a private organisation but no credentials have been specified? Are they also implementing a timeout or catching the prompt somehow?

CI servers will time out eventually.

> On the private API usage - Dependabot is trying to do dependency resolution and lockfile generation with as little additional work as possible here (i.e., without doing an install). Are there any plans to add a public API for that, similar to cargo update or npm install --package-lock-only?

There are no plans for this right now because I don't see what the use case is outside of dependabot, but I wouldn't be against contributions. We would need to decide if it should be added to Mix or Hex.

Reply:
> What is your goal with hex_erl?

The goal is to provide most of the infrastructure and helpers for tools like rebar3 and erlang.mk.